sys/phydat/phydat_str.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_sensif
+ * @{
+ *
+ * @file
+ * @brief       Generic sensor/actuator data handling
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+
+#include "phydat.h"
+
+void phydat_dump(phydat_t *data, int dim)
+{
+    if (data == NULL || dim < 0 || dim > PHYDAT_DIM) {
+        puts("Unable to display data object");
+        return;
+    }
+    printf("Data:");
+    for (int i = 0; i < dim; i++) {
```

Comment:
the maximum is `PHYDAT_DIM`. `dim` as paramter is used so you can print only a subset, e.g. 1-3, of the values - as not always all of them are populated